akka-contrib/src/multi-jvm/scala/akka/contrib/pattern/ClusterClientSpec.scala

```diff
@@ -244,5 +249,48 @@ class ClusterClientSpec extends MultiNodeSpec(ClusterClientSpec) with STMultiNod
       enterBarrier("after-5")
     }
 
+    "re-establish connection to receptionist after server restart" in within(30 seconds) {
+      runOn(client) {
+        remainingServerRoleNames.size should ===(1)
+        val remainingContacts = remainingServerRoleNames.map { r ⇒
+          system.actorSelection(node(r) / "user" / "receptionist")
+        }
+        val c = system.actorOf(ClusterClient.props(remainingContacts), "client4")
+
+        c ! ClusterClient.Send("/user/service2", "bonjour4", localAffinity = true)
+        expectMsg(10.seconds, "bonjour4-ack")
+        val lastSenderAddress = lastSender.path.address
+        val receptionistRoleName = roleName(lastSenderAddress) match {
+          case Some(r) ⇒ r
+          case None    ⇒ fail("unexpected missing roleName: " + lastSender.path.address)
+        }
+        receptionistRoleName should ===(remainingServerRoleNames.head)
+        val logSource = s"${system.asInstanceOf[ExtendedActorSystem].provider.getDefaultAddress}/user/client4"
+
+        EventFilter.info(start = "Connected to", source = logSource, occurrences = 1) intercept {
+          EventFilter.info(start = "Lost contact", source = logSource, occurrences = 1) intercept {
+            // shutdown server
+            testConductor.shutdown(remainingServerRoleNames.head).await
+          }
+        }
+
+        c ! ClusterClient.Send("/user/service2", "shutdown", localAffinity = true)
+        Thread.sleep(2000) // to ensure that it is sent out before shutting down system
+      }
+
+      runOn(remainingServerRoleNames.toSeq: _*) {
+        system.awaitTermination(20.seconds)
+        // start new system on same port
+        val sys2 = ActorSystem(system.name,
```

Comment:
Just add a comment please (I realized L254 already asserted that it is a single one), because it is not intuitive.